content/en/blog/2024/target-allocator-troubleshooting/index.md

```diff
@@ -0,0 +1,481 @@
+use it, like this:
+
+```yaml
+---
+prometheusCR:
+  enabled: true
+  podMonitorSelector: {}
+  serviceMonitorSelector: {}
+```
+
+See full example
+[here](https://github.com/avillela/otel-target-allocator-talk/blob/4c0eb425c90187d584c9d03b51ad918b377014a3/src/resources/02-otel-collector.yml#L15-L17).
+
+I just learned this today, as I was updating my `OpenTelemetryCollector` YAML
+from
+[`v1alpha1`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollector)
+to
+[`v1beta1`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollector-1).
+
+### 6- Do your labels, namespaces, and ports match for your ServiceMonitor and your Service (or PodMonitor and your Pod)?
+
+The `ServiceMonitor` is configured to pick up Kubernetes
+[Services](https://kubernetes.io/docs/concepts/services-networking/service/)
+that match on:
+
+- Labels
+- Namespaces (optional)
+- Ports (endpoints)
+
+Suppose that you have this `ServiceMonitor`:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: sm-example
+  labels:
+    app: my-app
+    release: prometheus
+spec:
+  selector:
+    matchLabels:
+      app: my-app
+  namespaceSelector:
+    matchNames:
+      - opentelemetry
+  endpoints:
+    - port: prom
+      path: /metrics
+    - port: py-client-port
+      interval: 15s
+    - port: py-server-port
+```
+
+The above`ServiceMonitor` is looking for any services that have:
+
+- the label `app: my-app`
+- reside in a namespace called `opentelemetry`
+- a port named `prom`, `py-client-port`, _or_ `py-server-port`
+
+So for example, the `Service` resource below would get picked up by the
+`ServiceMonitor`, because it matches the above criteria:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: py-prometheus-app
+  namespace: opentelemetry
+  labels:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+spec:
+  selector:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+  ports:
+    - name: prom
+      port: 8080
+```
+
+Conversely, the `Service` resource below would NOT, because the `ServiceMonitor`
+is looking for ports named `prom`, `py-client-port`, _or_ `py-server-port`, and
+this service’s port is called `bleh`.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: py-prometheus-app
+  namespace: opentelemetry
+  labels:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+spec:
+  selector:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+  ports:
+    - name: bleh
+      port: 8080
+```
+
+> **NOTE:** If you’re using `PodMonitor`, the same applies, except that it picks
+> up Kubernetes _pods_ that match on labels, namespaces, and named ports. You
+> can see an example `PodMonitor` resource definition
+> [here](https://github.com/avillela/otel-target-allocator-talk/blob/main/src/resources/04a-pod-monitor.yml).
```

Comment:
Ditto my comment on the previous call-out. Also, for links, we try to use more descriptive words than "here" when possible.

```suggestion
> **NOTE:** If you’re using `PodMonitor`, the same applies, except that it picks
> up Kubernetes _pods_ that match on labels, namespaces, and named ports. For example, see this `PodMonitor` [resource definition](https://github.com/avillela/otel-target-allocator-talk/blob/main/src/resources/04a-pod-monitor.yml).
```